pkg/envtest/server.go

```diff
@@ -62,14 +62,17 @@ const (
 	defaultKubebuilderControlPlaneStopTimeout  = 20 * time.Second
 )
 
-// Default binary path for test framework
-func defaultAssetPath(binary string) string {
-	assetPath := os.Getenv(envKubebuilderPath)
-	if assetPath == "" {
-		assetPath = defaultKubebuilderPath
+// getBinAssetPath will return the path for the binary informed
+func (te *Environment) getBinAssetPath(binary string) string {
+	if hasBinary(binary, te.BinaryDirectoryPath) {
+		return filepath.Join(te.BinaryDirectoryPath, binary)
 	}
-	return filepath.Join(assetPath, binary)
 
+	valueFromEnvVar := os.Getenv(envKubebuilderPath)
+	if hasBinary(binary, valueFromEnvVar) {
```

Comment:
I would probably throw in a `valueFromEnvVar != "" &&` on this, just to avoid silently using `/kube-apiserver`, though I admit it's unlikely that would exist :)

Reply:
(and same above)

Reply:
Alternatively could do `if path == "" { return false }` down in hasBinary.

Reply:
done in the hasBinary func 👍